### PR TITLE
Allow to open file in the Theia IDE from a sidecar container

### PR DESCRIPTION
**Pull request by tolusha**
### What does this PR do?
Allows to open file in the Theia IDE from a sidecar container.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/13614

### CODE REVIEW

**Comment by benoitf** [comment]
could you please fix the conflict ?

File: extensions/eclipse-che-theia-plugin-remote/src/node/document-container-aware.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    overrideGetDocumentData(documentExt: DocumentsExtImpl) {
+        const originalGetDocumentData = documentExt.getDocumentData.bind(documentExt);
+        const getDocumentData = (resource: theia.Uri) => originalGetDocumentData(this.overrideUri(resource));
+        documentExt.getDocumentData = getDocumentData;
+    }
+
+    private overrideUri(uri: URI | theia.Uri) {
+        if (!uri.path.startsWith('/projects')) {
```
Comment (benoitf) [inline extensions/eclipse-che-theia-plugin-remote/src/node/document-container-aware.ts]: sometimes it's not /projects how do we handle that ? is it provided in env variable as well ?

File: extensions/eclipse-che-theia-plugin-remote/src/node/document-container-aware.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            const newScheme = 'file-sidecar-' + process.env.CHE_MACHINE_NAME;
```
Comment (benoitf) [inline extensions/eclipse-che-theia-plugin-remote/src/node/document-container-aware.ts]: somehow the constant 'file-sidecar-' should be shared

File: extensions/eclipse-che-theia-plugin-remote/src/node/languages-container-aware.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        if ('uri' in reference) {
+            reference.uri = this.overrideUri(reference.uri);
+        } else {
+            reference.targetUri = this.overrideUri(reference.targetUri);
+        }
+    }
+
+    overrideUri(uri: URI | theia.Uri) {
+        if (!uri.path.startsWith('/projects')) {
+            const newScheme = 'file-sidecar-' + process.env.CHE_MACHINE_NAME;
```
Comment (benoitf) [inline extensions/eclipse-che-theia-plugin-remote/src/node/languages-container-aware.ts]: '/projects' shouldn't be hardcoded

File: extensions/eclipse-che-theia-plugin-remote/src/node/document-container-aware.ts (hunk 2 above)
Comment (tolusha) [reply extensions/eclipse-che-theia-plugin-remote/src/node/document-container-aware.ts]: It is possible to share constant between several extensions?

File: che-theia-init-sources.yml [hunk 4]
```diff
@@ -19,4 +19,4 @@ sources:
     - plugins/task-plugin
     - plugins/welcome-plugin
     - plugins/ssh-plugin
-  checkoutTo: master
+  checkoutTo: ab/scc
```
Comment (benoitf) [inline che-theia-init-sources.yml]: should be reverted, correct ?

**Comment by dmytro-ndp** [comment]
crw-ci-test

**Comment by dmytro-ndp** [comment]
new-crw-ci-test

**Comment by dmytro-ndp** [comment]
new-crw-ci-test

**Comment by dmytro-ndp** [comment]
new-crw-ci-test

**Comment by tolusha** [comment]
Tested with go, python and java.

**Comment by tolusha** [comment]
ci-build

**Comment by tolusha** [comment]
crw-ci-test

**Pull request merged**